ui/core/src/model/query.ts

```diff
@@ -12,16 +12,35 @@
 // limitations under the License.
 
 import { Definition, UnknownSpec } from './definitions';
+import { TimeSeriesData } from './time-series-data';
 
 interface QuerySpec<PluginSpec> {
   plugin: Definition<PluginSpec>;
 }
 /**
  * A generic query definition interface that can be extended to support more than just TimeSeriesQuery
  */
-export interface QueryDefinition<Kind = QueryType, PluginSpec = UnknownSpec> {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface QueryDefinition<Kind = any, PluginSpec = UnknownSpec> {
   kind: Kind;
   spec: QuerySpec<PluginSpec>;
 }
 
-export type QueryType = 'TimeSeriesQuery';
+/**
+ * Mapping the query plugin to the data type it returns
+ */
+export interface QueryType {
```

Comment:
This allows typescript to be able to infer the return type of 'useDataQueries` given the parameter type.